src/hyperlight_host/src/hypervisor/mod.rs

```diff
@@ -158,11 +161,9 @@ pub(crate) trait Hypervisor: Debug + Send {
         host_funcs: &Arc<Mutex<FunctionRegistry>>,
     ) -> Result<()>;
 
-    /// Run the vCPU
-    fn run(
-        &mut self,
-        #[cfg(feature = "trace_guest")] tc: &mut crate::sandbox::trace::TraceContext,
-    ) -> Result<HyperlightExit>;
+    /// Runs the vCPU until it exits.
+    /// Note: this function should not emit any traces or spans as it is called after guest span is setup
```

Comment:
This comment suggests to me that the guest span is being setup too early.

Reply:
OK lets create an issue? I think its more important that we don't rely on people not making mistakes than we do on having ideal APIs